activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -29,8 +36,35 @@ def create_table(table_name, options = {})
             options[:auto_increment] = true
           end
 
-          super
+          if block_given?
+            super(table_name, options) do |t|
+              class << t
+                prepend TableDefinition
```

Comment:
This looks identical to the body of the superclass definition. Am I missing something?